### PR TITLE
Update Algotrading_Advanced TA Backtesting.py

### PR DESCRIPTION
**Pull request by raghavdixit99**
---
# EntelligenceAI PR Summary 
 ### Release Note

**Purpose**:
- Enhance trading strategy with a trailing stop mechanism and comprehensive performance metrics.

**Changes**:
- **Enhancement**: 
  - `generate_signals_advanced` now includes moving average conditions for refined buy/sell signals.
  - `execute_strategy_with_stop_loss` upgraded with a trailing stop feature for dynamic stop-loss adjustments.
  - `calculate_performance_metrics` computes Sortino Ratio and win rate for detailed strategy evaluation.

**Impact**:
- Improved trading strategy accuracy, enhanced risk management, and a comprehensive view of strategy performance.

### CODE REVIEW

File: Algotrading_Advanced TA Backtesting.py [hunk 1]
```diff
@@ -47,51 +42,58 @@ def execute_strategy_with_stop_loss(df, slippage=0.0002, transaction_cost=0.001,
             if stop_loss is None:
                 stop_loss = data.loc[i - 1, "Close"] - atr_multiplier * atr
                 take_profit = data.loc[i - 1, "Close"] + atr_multiplier * atr
+                trailing_stop_level = stop_loss if trailing_stop else None
+
+            if trailing_stop and data.loc[i, "Close"] > take_profit * 0.5:
+                trailing_stop_level = max(trailing_stop_level, data.loc[i, "Close"] - atr)
 
-            if data.loc[i, "Close"] <= stop_loss or data.loc[i, "Close"] >= take_profit:
+            if data.loc[i, "Close"] <= stop_loss or (trailing_stop and data.loc[i, "Close"] <= trailing_stop_level):
```
Comment (aiswaryasankarwork) [inline Algotrading_Advanced TA Backtesting.py:50]: Missing handling of `take_profit` level in the stop condition check. Trade should also exit when take profit is hit even with trailing stop enabled.


<details>
<summary><strong>📝 Committable Code Suggestion</strong></summary>

> ‼️ Ensure you review the code suggestion before committing it to the branch. Make sure it replaces the highlighted code, contains no missing lines, and has no issues with indentation. 


```suggestion
            if data.loc[i, "Close"] <= stop_loss or data.loc[i, "Close"] >= take_profit or (trailing_stop and data.loc[i, "Close"] <= trailing_stop_level):
```
</details>

Comment (aiswaryasankarwork) [inline Algotrading_Advanced TA Backtesting.py:50]: Missing handling of `take_profit` level in the stop condition check. Trade can continue indefinitely if trailing stop is not hit.


<details>
<summary><strong>📝 Committable Code Suggestion</strong></summary>

> ‼️ Ensure you review the code suggestion before committing it to the branch. Make sure it replaces the highlighted code, contains no missing lines, and has no issues with indentation. 


```suggestion
            if data.loc[i, "Close"] <= stop_loss or data.loc[i, "Close"] >= take_profit or (trailing_stop and data.loc[i, "Close"] <= trailing_stop_level):
```
</details>

Comment (aiswaryasankarwork) [inline Algotrading_Advanced TA Backtesting.py:50]: Missing handling of `take_profit` level in the stop condition check. Trade should also exit when take profit is hit even with trailing stop enabled.


<details>
<summary><strong>📝 Committable Code Suggestion</strong></summary>

> ‼️ Ensure you review the code suggestion before committing it to the branch. Make sure it replaces the highlighted code, contains no missing lines, and has no issues with indentation. 


```suggestion
            if data.loc[i, "Close"] <= stop_loss or data.loc[i, "Close"] >= take_profit or (trailing_stop and data.loc[i, "Close"] <= trailing_stop_level):
```
</details>

Comment (aiswaryasankarwork) [inline Algotrading_Advanced TA Backtesting.py:50]: Missing handling of `take_profit` level in the stop condition check. Trade can continue indefinitely if trailing stop is not hit.


<details>
<summary><strong>📝 Committable Code Suggestion</strong></summary>

> ‼️ Ensure you review the code suggestion before committing it to the branch. Make sure it replaces the highlighted code, contains no missing lines, and has no issues with indentation. 


```suggestion
            if data.loc[i, "Close"] <= stop_loss or data.loc[i, "Close"] >= take_profit or (trailing_stop and data.loc[i, "Close"] <= trailing_stop_level):
```
</details>

Comment (aiswaryasankarwork) [inline Algotrading_Advanced TA Backtesting.py:50]: Missing handling of `take_profit` level in the stop condition check. Trade can continue indefinitely if trailing stop is not hit.


<details>
<summary><strong>📝 Committable Code Suggestion</strong></summary>

> ‼️ Ensure you review the code suggestion before committing it to the branch. Make sure it replaces the highlighted code, contains no missing lines, and has no issues with indentation. 


```suggestion
            if data.loc[i, "Close"] <= stop_loss or data.loc[i, "Close"] >= take_profit or (trailing_stop and data.loc[i, "Close"] <= trailing_stop_level):
```
</details>

Comment (aiswaryasankarwork) [inline Algotrading_Advanced TA Backtesting.py:50]: Missing handling of `take_profit` level in the stop condition check. Trade can continue indefinitely if trailing stop is not hit.


<details>
<summary><strong>📝 Committable Code Suggestion</strong></summary>

> ‼️ Ensure you review the code suggestion before committing it to the branch. Make sure it replaces the highlighted code, contains no missing lines, and has no issues with indentation. 


```suggestion
            if data.loc[i, "Close"] <= stop_loss or data.loc[i, "Close"] >= take_profit or (trailing_stop and data.loc[i, "Close"] <= trailing_stop_level):
```
</details>
